### PR TITLE
Winner window

### PR DESCRIPTION
**Pull request by luis-tm**
- modified lobby to fetch data when receiving socket message
- updated ready state to false when the owner navigates to the update lobby view
- added marks on the units to indicate they can be moved (for the moment, since it's better for testing, I've commented the line which allows units to move only once)
- added popup when clicking the surrender flag
- added popup with the "winner window"

(TODO: integrate all possible buttons with the server)

### CODE REVIEW

**Comment by luis-tm** [comment]
image
image
image

File: src/components/views/lobby/Lobby.js [hunk 1]
```diff
@@ -162,32 +162,36 @@ const Lobby = ({ id }) => {
...
+            setGetDataFailed(true);
+        }
+    }
+
+    useEffect(() => {
+        async function fetchData() {
+            obtainAndLoadLobbyInfo();
         }
 
         fetchData();
```
Comment (apple00juice) [inline src/components/views/lobby/Lobby.js:197]: can't you call directly obtainAndLoadLobbyInfo ?

File: src/components/fragments/game/unit/Unit.js [hunk 2]
```diff
@@ -70,6 +82,9 @@ const Unit = props => {
             <UnitImage type={props.unit.type} color={unitColor} animation={animationState}/>
             <img src={healthSource} style={healthBoxStyle} alt={''}/>
             <div style={healthBarStyle} />
+            {
+                props.unit.teamId == myTeam && props.unit.selected == false ? <img src={markerSource} style={markerStyle} alt={''} /> : null
```
Comment (apple00juice) [inline src/components/fragments/game/unit/Unit.js:86]: I think you also have to test if commands is empty.
e.g. if commands is empty the unit has already performed its moves and thus shouldn't have a marker

**Review by apple00juice — APPROVED** [review]
Rest looks good 👍

File: src/components/fragments/game/unit/Unit.js (hunk 2 above)
Comment (luis-tm) [reply src/components/fragments/game/unit/Unit.js:86]: ah ok! I was using "selected" to check it (since we were not using it, I change it to set the field = true when the unit has moved or attacked). I'll change it.

Comment (apple00juice) [reply src/components/fragments/game/unit/Unit.js:86]: I wouldn't show it, if the unit is selected too. As it is probably to many UI elements. But you have to have to check what looks better.

Comment (apple00juice) [reply src/components/fragments/game/unit/Unit.js:86]: > ah ok! I was using "selected" to check it (since we were not using it, I change it to set the field = true when the unit has moved or attacked). I'll change it.

Ahh, now I understand. But we set selected back to false after the movement right?

Comment (luis-tm) [reply src/components/fragments/game/unit/Unit.js:86]: I changed it. Now when "cleaning" the units the unit.selected is not set to false again.

Comment (apple00juice) [reply src/components/fragments/game/unit/Unit.js:86]: Then this also works, but I would change the variable to something like performedAction

**Review by mkorob — APPROVED** [review]
